Doc/reference/datamodel.rst

```diff
@@ -2616,6 +2616,23 @@ are awaitable.
    :term:`awaitable` objects.  For instance, :class:`asyncio.Future` implements
    this method to be compatible with the :keyword:`await` expression.
 
+   .. note::
+
+      The language doesn't place any restrictions on the type or value of the
+      objects yielded by the iterator returned by ``__await__``, as this is
+      specific to the implementation of the asynchronous execution framework
+      (e.g. :mod:`asyncio`) that will be managing the :term:`awaitable` object.
+
+      In the case of :mod:`asyncio`, user code should generally be using
+      :term:`coroutines <coroutine>`, instances of :class:`asyncio.Task`,
+      :class:`asyncio.Future`, and other :mod:`asyncio` objects to implement
+      ``__await__``; yielding objects from these (e.g.
+      ``return (yield from future)``), rather than yielding objects directly.
+      An exception to this is a ``yield None`` (or equivalent bare ``yield``)
```

Comment:
I don't think the special case of `await asyncio.sleep(0)` via `yield None` should be documented here

Reply:
Agreed.